app/services/foreman_rh_cloud/template_renderer_helper.rb

```diff
@@ -15,7 +15,16 @@ module TemplateRendererHelper
     end
     def remediations_playbook(hit_remediation_pairs)
       hit_remediation_pairs = JSON.parse(hit_remediation_pairs)
-      retriever = ForemanRhCloud::RemediationsRetriever.new(hit_remediation_pairs, logger: template_logger)
+      retriever = ForemanRhCloud::HitRemediationsRetriever.new(hit_remediation_pairs, logger: template_logger)
+      retriever.create_playbook
+    end
+
+    apipie :method, 'Returns a Red Hat remediation playbook compiled on console.redhat.com' do
+      required :remediation_path, String, desc: ''
+      returns String, desc: 'Playbook downloaded from the cloud'
+    end
+    def download_rh_playbook(playbook_url)
```

Comment:
I'd suggest an early return if the render is done as a preview. Also it would be a good idea to cache the playbook to avoid re-downloading the same playbook for each host separately.

We already did that for the first iteration of cloud integration https://github.com/theforeman/foreman_ansible/blob/master/app/services/foreman_ansible/renderer_methods.rb#L20-L22